### ISSUE TITLE
UI error 

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#43335]
- **Sourcegraph version:** Sourcegraph.com
- **Platform information:** Chrome version 105.0.5195.125 (Official Build) (64-bit) on Linux
 
#### Steps to reproduce:

1. Visit https://sourcegraph.com/github.com/adobe/react-spectrum@main/-/blob/packages/dev/docs/package.json
2. Ensure the Git blame gutter is showing
3. Click on a blank line in the file
4. Click in the file tree
5. Switch focus away and then return focus
6. Wait about 3-4 seconds

#### Expected behavior:

No error

#### Actual behavior:

An error is displayed, and Git blame information is displayed in the top left. The error links to https://reactjs.org/docs/error-decoder.html/?invariant=185: 

```
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

![image](https://user-images.githubusercontent.com/1976/197408401-1433ef5c-7fd8-498e-8d07-360a15abae83.png)

![image](https://user-images.githubusercontent.com/1976/197408426-d73e9765-9e4a-4923-be1c-1eefb935d6da.png)

/cc @sourcegraph/integrations

**Comment by philipp-spiess** [comment]
So far I was only able to reproduce this in a production environment (i.e with minified symbols and the production React path). However I got the issue recoded in a [production DevTools performance trace](https://github.com/sourcegraph/sourcegraph/files/9851387/Profile-20221024T135244.json.zip).

**Comment by philipp-spiess** [comment]
@vovakulikov I think that your recent PR that improves the dashboard select UI has something to do with this regression: https://github.com/sourcegraph/sourcegraph/pull/43029

If you take a look into the DevTools trace I uploaded above, a lot of the time spent on the long running and eventually overflowing task is spent inside focus handler:

<img width="2558" alt="Screenshot 2022-10-24 at 14 08 20" src="https://user-images.githubusercontent.com/458591/197522114-e002ef89-c6bc-4030-83b9-5604b8e08502.png">

Specifically, this line of code seems to emit the events that are happening inside an endless loop:

<img width="2559" alt="Screenshot 2022-10-24 at 14 06 29" src="https://user-images.githubusercontent.com/458591/197522223-fc9853d9-3f9b-4ac8-a9de-d06dd8196aea.png">

I also tried to jump to the commit before your changes (`2a7215e485`) and could not reproduce the issue (I can reproduce locally now with `SOURCEGRAPH_API_URL=https://sourcegraph.com sg start web-standalone-prod` on the `main` branch).

Would you be up to pair on fixing this?

**Comment by philipp-spiess** [comment]
Oh @vovakulikov is out. @valerybugakov maybe you can help me out and pair with me on this? 😬 

Should we revert https://github.com/sourcegraph/sourcegraph/pull/43029 while we figure this out? (cc @AlicjaSuska)

**Comment by vovakulikov** [comment]
@philipp-spiess I think I know what happens there. 

Previously before we merged https://github.com/sourcegraph/sourcegraph/pull/43029 the PopoverContent didn't do anything with focus (didn't return it to the original target) if `focusLocked` is specified and set to `false` and also didn't autofocus anything within. In https://github.com/sourcegraph/sourcegraph/pull/43029 I changed this behaviour and now PopoverContent exposes the `returnTargetFocus` prop to turn on/off focus preservation after Popover has been closed. 

The recursion (why we have this problem) is 
- Blame line (let's call it target 1) is listening to focus and showing a tooltip on the focused target (let's call this Popover1) 
- If you move the cursor (hover over another blame line, let's say target 2) or you're going through blame lines with a keyboard you focus next target (target 2), and the previous target (target 1) emits a blur which closes the Popover 1 
- Since `returnTargetFocus` by default is true the Popover 1 returns popover to the target 1
- And this is a loop, you're hovering over target 2, and we close popover 1, which closes returns focus to target 1, and we open Popover 1 

I shared my feedback about this that this blame popover should use a tooltip pattern and doesn't have any interactive element as content. And in this case, we wouldn't be needed in any focus popover-like management in this UI. 

Here is a PR with the fix https://github.com/sourcegraph/sourcegraph
